deeposlandia/paramoptim.py

```diff
@@ -0,0 +1,298 @@
+
+def get_data(folders, dataset, model, image_size, batch_size):
+    """
+    """
+    # Data gathering
+    if (os.path.isfile(folders["training_config"]) and os.path.isfile(folders["validation_config"])
+        and os.path.isfile(folders["testing_config"])):
+        train_config = utils.read_config(folders["training_config"])
+        label_ids = [x['id'] for x in train_config['labels'] if x['is_evaluate']]
+        train_generator = generator.create_generator(
+            dataset,
+            model,
+            folders["training"],
+            image_size,
+            batch_size,
+            label_ids,
+            seed=SEED)
+        validation_generator = generator.create_generator(
+            dataset,
+            model,
+            folders["validation"],
+            image_size,
+            batch_size,
+            label_ids,
+            seed=SEED)
+        test_generator = generator.create_generator(
+            dataset,
+            model,
+            folders["testing"],
+            image_size,
+            batch_size,
+            label_ids,
+            inference=True,
+            seed=SEED)
+    else:
+        utils.logger.error(("There is no valid data with the specified parameters. "
+                           "Please generate a valid dataset before calling the training program."))
+        sys.exit(1)
+    nb_labels = len(label_ids)
+    return nb_labels, train_generator, validation_generator, test_generator
+
+def run_model(train_generator, validation_generator, output_folder,
+              instance_name, image_size, aggregate_value, nb_labels,
+              nb_training_image, nb_validation_image,
+              batch_size, dropout, network, learning_rate, learning_rate_decay):
+    """
+    """
+    if args.model == "feature_detection":
+        net = FeatureDetectionNetwork(network_name=instance_name,
+                                      image_size=image_size,
+                                      nb_channels=3,
+                                      nb_labels=nb_labels,
+                                      architecture=network)
+        loss_function = "binary_crossentropy"
+    elif args.model == "semantic_segmentation":
+        net = SemanticSegmentationNetwork(network_name=instance_name,
+                                          image_size=image_size,
+                                          nb_channels=3,
+                                          nb_labels=nb_labels,
+                                          architecture=network)
+        loss_function = "categorical_crossentropy"
+    else:
+        utils.logger.error(("Unrecognized model. Please enter 'feature_detection' "
+                            "or 'semantic_segmentation'."))
+        sys.exit(1)
+    model = Model(net.X, net.Y)
+    opt = Adam(lr=learning_rate, decay=learning_rate_decay)
+    model.compile(loss=loss_function, optimizer=opt, metrics=['acc'])
+
+    # Model training
+    STEPS = nb_training_image // batch_size
```

Comment:
I think this is not necessary to have variable names in uppercase.

Reply:
Both affected variables have been lower-cased.